feat(ios): update the Firebase iOS SDK to 11.15.0

Astrovic opened the pull request:
- Update Firebase iOS to v.10.25.0
- Added code signatures to all of Firebase's binary artifacts (https://github.com/firebase/firebase-ios-sdk/issues/12238).

| Download |
| ---------- | 
| [firebase.core-iphone-11.0.0.zip](https://github.com/Astrovic/titanium-firebase-core/releases/download/ios-11.0.0/firebase.core-iphone-11.0.0.zip) |

Astrovic committed:
feat(ios): update the Firebase iOS SDK to 10.25.0

hansemannn commented:
@Astrovic Thanks for the great line of updates! Just to summarize, which ones are covered:

- core
- performance
- cloud messaging
- in-app-messaging
- analytics
- crashlytics

I can update auth and remote-config if you want - the other missing ones are rather minor and not frequently used.

Astrovic commented:
Thanks, I only updated the modules I use in a single project:

| Module | Release |
| ---------- | ---------- |
|titanium-firebase-analytics | [iOS 10.0.0](https://github.com/Astrovic/titanium-firebase-analytics/releases/tag/ios-10.0.0)|
|titanium-firebase-analytics | [Android 5.3.0](https://github.com/Astrovic/titanium-firebase-analytics/releases/tag/android-5.3.0)|
| titanium-firebase-core | [iOS 11.0.0](https://github.com/Astrovic/titanium-firebase-core/releases/tag/ios-11.0.0) |
| titanium-crashlytics | [iOS 8.0.0](https://github.com/Astrovic/titanium-crashlytics/releases/tag/ios-8.0.0) |
| titanium-firebase-cloud-messaging | [iOS 11.0.0](https://github.com/Astrovic/titanium-firebase-cloud-messaging/releases/tag/ios-11.0.0) |
| titanium-firebase-in-app-messaging | [iOS 7.0.0](https://github.com/Astrovic/titanium-firebase-in-app-messaging/releases/tag/ios-7.0.0) |
| titanium-firebase-performance | [iOS 10.0.0](https://github.com/Astrovic/titanium-firebase-performance/releases/tag/ios-3.0.0) |

I moved all the frameworks to core and to the shared folder of each module that has shared frameworks, to avoid duplicate errors.

If there are other necessary frameworks, it is better to add them to the core, and then also put them in the shared folder, so in the future it will be easier to update them without risking conflicts due to duplicate frameworks.

hansemannn commented:
I actually tried to prevent exactly that, because some others may not want to use a certain framework, but it would still be signed and bundled, increasing the app size a lot. That's why it was so sensitively-maintained

Astrovic commented:
Yes, this was clear. On the core module I added only a few frameworks, and I added a /shared folder in the modules that had shared framewoks, following the instructions in the [METADATA.md](https://github.com/hansemannn/titanium-firebase-core/files/15306507/METADATA.md) file present in the Firebase SDK zip.

It is not clear to me whether this approach is correct, but I manually remove any unnecessary frameworks from the /platform folder, after unzipping the module zip file.

hansemannn commented:
You mean you removed frameworks that were there before? Then I have to revisit all PRs again: The `firebase-core` module includes all frameworks that are used by more than one other module, e.g. one thats used in both crashlytics and remote-config.

Astrovic commented:
No, I haven't removed any frameworks here in the core module. Core contains all the frameworks needed for Analytics, and some frameworks shared with the other Firebase modules. 
I tested the core module individually with all the other modules I updated.

I mean that before I build my apps, I sometimes remove unnecessary frameworks from the /platform folder of some modules to lighten the size of the app.

hansemannn commented:
Ahh okay, yeah thats possible and fine. I plan to merge all PR by end of the week (and add auth + remote config before), then we have all in one state :-) Thanks again!

Astrovic committed:
feat(ios): update the Firebase iOS SDK to 11.15.0

Astrovic commented:
I have updated these modules to Firebase iOS SDK 11.15.0, updating their PRs in your repos

| Module | Release |
| ---------- | ---------- |
|titanium-firebase-analytics | [iOS 11.15.0](https://github.com/Astrovic/titanium-firebase-analytics/releases/tag/ios-11.15.0)|
| titanium-firebase-core | [iOS 11.15.0](https://github.com/Astrovic/titanium-firebase-core/releases/tag/ios-11.15.0) |
| titanium-crashlytics | [iOS 9.0.0](https://github.com/Astrovic/titanium-crashlytics/releases/tag/ios-9.0.0) |
| titanium-firebase-cloud-messaging | [iOS 11.15.0](https://github.com/Astrovic/titanium-firebase-cloud-messaging/releases/tag/ios-11.15.0) |
| titanium-firebase-in-app-messaging | [iOS 8.0.0](https://github.com/Astrovic/titanium-firebase-in-app-messaging/releases/tag/ios-8.0.0) |
| titanium-firebase-performance | [iOS 4.0.0](https://github.com/Astrovic/titanium-firebase-performance/releases/tag/ios-4.0.0) |

hansemannn commented:
Thanks! The version look a bit odd, as we usually don't inherit the versions from Firebase but maintain own version (for example if we do proxy changes from 11.5.0, it's 11.5.1, but thats not the Firebase 11.5.1 version and we cannot use that version anymore then.

Therefore, in order to merge it to the main Firebase-project, it requires a new major update (x.0.0) in order to be merged. Is that possible? Thanks for your help!

Astrovic committed:
chore(ios): update the Firebase iOS SDK to 11.15.0

Astrovic commented:
Sure, I just updated the versions of the modules, here are all the links and the relative zips

| Module | Release | Download |
| ---------- | ---------- | ---------- |
|titanium-firebase-analytics | [iOS 12.0.0](https://github.com/Astrovic/titanium-firebase-analytics/releases/tag/ios-12.0.0)| [zip](https://github.com/Astrovic/titanium-firebase-analytics/releases/download/ios-12.0.0/firebase.analytics-iphone-12.0.0.zip)| 
| titanium-firebase-core | [iOS 12.0.0](https://github.com/Astrovic/titanium-firebase-core/releases/tag/ios-12.0.0) | [zip](https://github.com/Astrovic/titanium-firebase-core/releases/download/ios-12.0.0/firebase.core-iphone-12.0.0.zip)| 
| titanium-crashlytics | [iOS 9.0.0](https://github.com/Astrovic/titanium-crashlytics/releases/tag/ios-9.0.0) | [zip](https://github.com/Astrovic/titanium-crashlytics/releases/download/ios-9.0.0/ti.crashlytics-iphone-9.0.0.zip)| 
| titanium-firebase-cloud-messaging | [iOS 12.0.0](https://github.com/Astrovic/titanium-firebase-cloud-messaging/releases/tag/ios-12.0.0) | [zip](https://github.com/Astrovic/titanium-firebase-cloud-messaging/releases/download/ios-12.0.0/firebase.cloudmessaging-iphone-12.0.0.zip)| 
| titanium-firebase-in-app-messaging | [iOS 8.0.0](https://github.com/Astrovic/titanium-firebase-in-app-messaging/releases/tag/ios-8.0.0) | [zip]( https://github.com/Astrovic/titanium-firebase-in-app-messaging/releases/download/ios-8.0.0/firebase.inappmessaging-iphone-8.0.0.zip)|
| titanium-firebase-performance | [iOS 4.0.0](https://github.com/Astrovic/titanium-firebase-performance/releases/tag/ios-4.0.0) | [zip](https://github.com/Astrovic/titanium-firebase-performance/releases/download/ios-4.0.0/firebase.performance-iphone-4.0.0.zip)|

Astrovic commented:
@hansemannn I just want to remind you that I am using the updated modules in production without any problems, so you could accept the PRs, to facilitate the next updates :)

hansemannn commented:
Thanks! I will go through by tomorrow!

hansemannn commented:
All PR's merged except https://github.com/hansemannn/titanium-firebase-in-app-messaging/pull/8 which needs further attention to verify it doesn't crash. Once merged, I'll need a new binary of the analytics module, then I'll release all new versions. Thanks a lot, @Astrovic! :-)

Astrovic commented:
Just done it!
ASAP I will make new PRs and builds for the 16kb support

hansemannn commented:
Can you please update the list of zip's, so I can release all new versions? It got a bit hard to keep track with the last changes. And maybe we wait for the 16kb support modules? Let me know if @prashantsaini1 or I should help

m1ga commented:
> And maybe we wait for the 16kb support modules?

I would create them as early as you can as people get the warnings already (Did some here in this thread already: https://tidev.slack.com/archives/C03CVQX2A/p1755179151819569).
It can still be used by anyone else with 12.7.x or higher. Modules just have to build with the current main branch of the SDK and that's it.

Astrovic commented:
> Can you please update the list of zip's, so I can release all new versions? It got a bit hard to keep track with the last changes. 

The list with the latest iOS versions (Firebase iOS SDK 11.15.0) and their zip files is the last one in the comments just above:: https://github.com/hansemannn/titanium-firebase-core/pull/71#issuecomment-3027907292
| Module | Release | Download |
| ---------- | ---------- | ---------- |
|titanium-firebase-analytics | [iOS 12.0.0](https://github.com/Astrovic/titanium-firebase-analytics/releases/tag/ios-12.0.0)| [zip](https://github.com/Astrovic/titanium-firebase-analytics/releases/download/ios-12.0.0/firebase.analytics-iphone-12.0.0.zip)| 
| titanium-firebase-core | [iOS 12.0.0](https://github.com/Astrovic/titanium-firebase-core/releases/tag/ios-12.0.0) | [zip](https://github.com/Astrovic/titanium-firebase-core/releases/download/ios-12.0.0/firebase.core-iphone-12.0.0.zip)| 
| titanium-crashlytics | [iOS 9.0.0](https://github.com/Astrovic/titanium-crashlytics/releases/tag/ios-9.0.0) | [zip](https://github.com/Astrovic/titanium-crashlytics/releases/download/ios-9.0.0/ti.crashlytics-iphone-9.0.0.zip)| 
| titanium-firebase-cloud-messaging | [iOS 12.0.0](https://github.com/Astrovic/titanium-firebase-cloud-messaging/releases/tag/ios-12.0.0) | [zip](https://github.com/Astrovic/titanium-firebase-cloud-messaging/releases/download/ios-12.0.0/firebase.cloudmessaging-iphone-12.0.0.zip)| 
| titanium-firebase-in-app-messaging | [iOS 8.0.0](https://github.com/Astrovic/titanium-firebase-in-app-messaging/releases/tag/ios-8.0.0) | [zip]( https://github.com/Astrovic/titanium-firebase-in-app-messaging/releases/download/ios-8.0.0/firebase.inappmessaging-iphone-8.0.0.zip)|
| titanium-firebase-performance | [iOS 4.0.0](https://github.com/Astrovic/titanium-firebase-performance/releases/tag/ios-4.0.0) | [zip](https://github.com/Astrovic/titanium-firebase-performance/releases/download/ios-4.0.0/firebase.performance-iphone-4.0.0.zip)|

And this is the Firebase Analytics Android module updated to Firebase Android SDK 22.4.0: [firebase.analytics-android-5.4.0.zip](https://github.com/user-attachments/files/22005097/firebase.analytics-android-5.4.0.zip)


> And maybe we wait for the 16kb support modules? Let me know if prashantsaini1 or I should help

As @m1ga said, modules just have to build with the current main branch of the SDK (`ti sdk install -b master`) and that's it. I followed his instructions https://github.com/tidev/titanium-sdk/issues/14239:

> **Steps:**
> 
> * Use SDK 12.8.1 (master)
> * set modules minsdk to 12.7.0 (modules work with that version in case people want to use the new modules already)
> * rebuild the module: `ti build -p android -b --sdk 12.8.1`
> * add them to an app and use [this script](https://cs.android.com/android/platform/superproject/main/+/main:system/extras/tools/check_elf_alignment.sh) to run `check_elf_alignment.sh file.apk` and see if all `arm64-v8a` lines say `ALIGNED`
